apps/web/components/PaymentButtons/PaymentButtons.vue

```diff
@@ -126,7 +126,20 @@ const paypalApplePayPaymentId = computed(() => {
   return paymentProviderGetters.getIdByPaymentKey(paymentMethods.value.list, PayPalApplePayKey);
 });
 
-const handleReadyToBuy = (callback?: PayPalAddToCartCallback) => {
+const handlePreparePayment = async (callback?: PayPalAddToCartCallback) => {
+  await doAdditionalInformation({
```

Comment:
This way we do the call first and then, and then do the ready checks, I would say there is no reason for doing that right? If the ready checks fail, we dont send the call.